wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -61,7 +79,39 @@ public class CacheConnection {
           "true",
           "Whether to use SSL for cache connections.");
 
+  protected static final AwsWrapperProperty CACHE_IAM_REGION =
+      new AwsWrapperProperty(
+          "cacheIamRegion",
+          null,
+          "AWS region for ElastiCache IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_IAM_USERNAME =
+      new AwsWrapperProperty(
+          "cacheIamUsername",
+          null,
+          "Username for ElastiCache IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_USERNAME =
+      new AwsWrapperProperty(
+          "cacheUsername",
+          null,
+          "Username for ElastiCache regular authentication.");
+
+  protected static final AwsWrapperProperty CACHE_PASSWORD =
+      new AwsWrapperProperty(
+          "cachePassword",
+          null,
+          "Password for ElastiCache regular authentication.");
+
+  protected static final AwsWrapperProperty CACHE_NAME =
+      new AwsWrapperProperty(
+          "cacheName",
+          null,
+          "Explicit cache name for ElastiCache IAM authentication. ");
+
   private final boolean useSSL;
+  private final boolean iamAuthEnabled;
+  private final Properties connectionProperties = new Properties();
```

Comment:
Why do you need to store a copy of all the properties? I would only store the relevant ones.